examples/advanced/rag/embedding/README.md

```diff
@@ -0,0 +1,94 @@
+
+![iter_single](./figs/iter_single.png)
+
+### Adaptation Step 2: federated with NVFlare
+Once we have the iterative training script ready with "starting model" loading capability, it can be easily adapted to a NVFlare trainer by using [Client API](../../../hello-world/ml-to-fl/pt/README.md).
+
+The major code modifications are for receiving the global model, set it as the starting point for each round's training, and returning the trained model after each local training round.
+
+## Job for NVFlare FL Training
+With the local training script ready, we can go ahead to generate the NVFlare job configs by reusing the job templates.
+
+Let's set the job template path with the following command.
+```bash
+nvflare config -jt ./job_template/
+```
+Then we can check the available templates with the following command.
+```bash
+nvflare job list_templates
+```
+We can see the "sag_pt_deploy_map" template is available, with which we further generate job configs for embedding model training as:
+```
+nvflare job create -force \
+  -j "/tmp/embed/nvflare/job" -w "sag_pt_deploy_map" -sd "code" \
+  -f meta.conf min_clients=3 \
+  -f app_1/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name nli" \
+  -f app_2/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name squad" \
+  -f app_3/config_fed_client.conf app_script="train_fl.py" app_config="--dataset_name quora" \
+  -f app_server/config_fed_server.conf model_class_path="st_model.SenTransModel" components[0].args.model.args.model_name="microsoft/mpnet-base" min_clients=3 num_rounds=7 key_metric="eval_loss" negate_key_metric=True 
+```
+
+
+For both client and server configs, we only set the necessary task-related parameters tasks, and leave the rest to the default values.
+
+## Federated Training
+With the produced job, we run the federated training on a single client using NVFlare Simulator.
+```
+nvflare simulator -w /tmp/embed/nvflare/workspace -n 3 -t 3 /tmp/embed/nvflare/job
+```
+
+## Results
+The evaluation on two test datasets - [stsb](https://huggingface.co/datasets/sentence-transformers/stsb) with embedding similarity evaluation, and [NLI](https://huggingface.co/datasets/sentence-transformers/all-nli) with triplet accuracy evaluation, are shown below.
+
+ TrainData | STSB_pearson_cos | STSB_spearman_euc | NLI_cos_acc | NLI_euc_acc
+--- |------------------|-------------------|-------------| ---
+NLI | 0.7586           | 0.7895            | 0.8033      | 0.8045
+Squad | 0.8206           | 0.8154            | 0.8051      | 0.8042
+Quora | 0.8161           | 0.8121            | 0.7891      | 0.7854
+All | 0.8497           | 0.8523            | 0.8426      | 0.8384
+Federated | 0.8444           | 0.8368            | 0.8269      |  0.8246
+
+As shown, the federated training results are better than individual site's, and can be close to the centralized training results, demonstrating the effectiveness of NVFlare in embedding model tuning tasks.
```

Comment:
Could you add a little more explanation to this paragraph?

1. Include links to "SQUAD", "QUORA" dataset?
2. Explain left hand side, do you mean: 

NLI -> Single site training using NLI data

Squad -> Single site training using Squad data

Quora -> Single site training using Quora data

All -> Single site training using NLI, Squad and Quora data, do you just use them all or there is any ratio to mix

Federated -> Three sites where each site contains its own data of NLI, Squad or Quora